app/llx/app/src/main/java/net/pierrox/lightning_launcher/feature/settings/SettingsFragment.kt

```diff
@@ -0,0 +1,186 @@
+
+        val pc = preferenceScreen.findPreference(KEY_CAT_INFOS) as PreferenceCategory?
+        val ratePreference = findPreference<Preference>(KEY_RATE)
+        val upgradePreference = findPreference<Preference>(KEY_UPGRADE)
+
+        if (!Version.HAS_RATE_LINK || app.isFreeVersion() || app.isTrialVersion()) {
+            ratePreference?.let { pc?.removePreference(it) }
+        }
+        if (!app.isFreeVersion() && !app.isTrialVersion()) {
+            upgradePreference?.let { pc?.removePreference(it) }
+        }
+    }
+
+    private fun setupPreference(key: String, title: Int, summary: Int) {
+        val p: Preference? = findPreference(key)
+        if (key == KEY_CAT_INFOS) {
+            try {
+                val pi: PackageInfo =
+                    requireContext().packageManager.getPackageInfo(requireContext().packageName, 0)
+                p?.title = getString(title) + "  v" + pi.versionName
+            } catch (e: PackageManager.NameNotFoundException) {
+                // pass, cannot fail
+            }
+        } else {
+            p?.setTitle(title)
+        }
+        if (summary != 0) p!!.setSummary(summary)
+        p?.onPreferenceClickListener = this
+    }
+
+    override fun onPreferenceClick(preference: Preference): Boolean {
+        val key: String = preference.key
+
+        if (KEY_CONFIGURE_PAGES == key) {
+            val intent = Intent(activity, ScreenManager::class.java)
+            intent.setAction(Intent.ACTION_EDIT)
+            startActivity(intent)
+        } else if (KEY_BACKUP_RESTORE == key) {
+            startActivity(Intent(activity, BackupRestore::class.java))
+        } else if (KEY_SELECT_LAUNCHER == key) {
+            PhoneUtils.selectLauncher(activity, true)
+        } else if (KEY_UPGRADE == key) {
+            LLApp.get().startUnlockProcess(activity)
+        } else if (KEY_TEMPLATES_BROWSE == key) {
+            startActivity(
+                Intent.createChooser(
+                    Intent(
+                        Intent.ACTION_VIEW,
+                        Version.BROWSE_TEMPLATES_URI
+                    ), ""
+                )
+            )
+        } else if (KEY_TEMPLATES_APPLY == key) {
+            launchApplyTemplates()
+        } else if (KEY_RATE != key && KEY_COMMUNITY != key) {
+            val intent = Intent(activity, Customize::class.java)
+            var path: ContainerPath? = null
+            if (KEY_CURRENT_PAGE == key) {
+                val current_page = LLApp.get().appEngine.readCurrentPage(Page.FIRST_DASHBOARD_PAGE)
+                val s: String =
+                    requireActivity().intent?.getStringExtra(Customize.INTENT_EXTRA_PAGE_PATH)
+                        .orEmpty()
+                if (s == null) {
+                    path = ContainerPath(
+                        requireActivity().intent?.getIntExtra(
+                            Customize.INTENT_EXTRA_PAGE_ID,
+                            current_page
+                        ) ?: 0
```

Comment:
Even though this should never happen (the original code would nullPointer-crash) better to leave the expected default, just in case.

```suggestion
                        ) ?: current_page
```